x-pack/test/functional/apps/observability/alerts/index.ts

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import expect from '@kbn/expect';
+import querystring from 'querystring';
+import { FtrProviderContext } from '../../../ftr_provider_context';
+
+// Based on the x-pack/test/functional/es_archives/observability/alerts archive.
+const DATE_WITH_DATA = {
+  rangeFrom: '2021-08-23T13:36:22.109Z',
+  rangeTo: '2021-08-25T13:36:22.109Z',
+};
+
+export default ({ getPageObjects, getService }: FtrProviderContext) => {
+  const esArchiver = getService('esArchiver');
+
+  describe('Observability alerts', function () {
+    this.tags('includeFirefox');
+
+    const pageObjects = getPageObjects(['common']);
+    const testSubjects = getService('testSubjects');
+
+    before(async () => {
+      await esArchiver.load('x-pack/test/functional/es_archives/observability/alerts');
+      await pageObjects.common.navigateToUrlWithBrowserHistory(
+        'observability',
+        '/alerts',
+        `?${querystring.stringify(DATE_WITH_DATA)}`
+      );
+    });
+
+    after(async () => {
+      await esArchiver.unload('x-pack/test/functional/es_archives/observability/alerts');
+    });
+
+    describe('Alerts table', () => {
+      it('Renders the table', async () => {
+        await testSubjects.existOrFail('events-viewer-panel');
+      });
+
+      it('Renders the correct number of cells', async () => {
+        // NOTE: This isn't ideal, but EuiDataGrid doesn't really have the concept of "rows"
+        const cells = await testSubjects.findAll('dataGridRowCell');
```

Comment:
Does EUI provide some kind of POM/Facades for their components? 

Else we might consider in the future creating something like that for ourself. Using selectors that another team specifies could easily lead to our tests breaking and having some layer in between would make it easier to fix.

Reply:
I don't think so.

I agree using selectors from other teams is generally not a good idea. However, in this case (and I'm not sure if this is clear at first) the `testSubjects` service hooks into our `data-test-subj` attributes (as opposed to random classes / IDs etc), this is a well established pattern in Kibana. Generally speaking EUI don't randomly change the top level `data-test-subj` attributes on their components. I did look at the time to see if the cells could take a custom `data-test-subj` as a prop, but it didn't seem to be the case (passing this as a prop to EUI components is very often an option). I might have missed the option somewhere, though.

Reply:
I see, thanks for the info!

Reply:
The other benefit is that if EUI breaks our tests with a PR, they get to fix them :D